Issue#202 Reuse query service for getOne operation

tex1988 opened the pull request:
dev
## ZenHub

* [Main ZenHub ticket](https://app.zenhub.com/workspaces/home-project-5f7b77ff8db73a001cb17009/issues/ita-social-projects/home/202)


## Code reviewers

- [ ]  @@kh0ma, @VadymSokorenko, @denis-litvinov, @likeRewca, @Abhai2016, @Ilya-Ross, @annaprok

### Second Level Review

- [ ] @github_username

## Summary of issue

The existing functionality was not reused

## Summary of change

- added getOne method to QueryApiService
- changed controllers to use getOne method of QueryApiService for getOne operations
- deleted getOne methods in entity services as unused
- fixed tests according to new exception messages

## Testing approach

Automation

## CHECK LIST
- [x]  СI passed
- [x]  Сode coverage >=95%
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [x]  All reviewers agreed to merge the PR
- [x]  I've checked new feature as logged in and logged out user if needed
- [x]  PR meets all conventions

tex1988 committed:
Issue#202: Changes in openapi.yaml
- changed names of id path params

tex1988 committed:
Issue#202: Changes in QueryApiService
- changed getParameterValue method to return values of path parameters also
- changed getPageFromQuery method to throw NotFoundHomeException when returned Page is empty
- added getOne method to add ability to use QueryApiService for getting one element by id
- changed javadoc

tex1988 committed:
Issue#202: Changes in CommonApi
- deleted entitySpecificationService field as unused
- added protected modifier for buildQueryResponse method

tex1988 committed:
Issue#202: Changes in CooperationApiImpl, NewsApiImpl, UserApiImpl
- changed methods that get one element by id to use QueryApiService

tex1988 committed:
Issue#202: Changes in QueryContactIT, QueryCooperationIT
- fixed test that was passed because of lack of validation on service layer

tex1988 committed:
Issue#202: Changes in QueryApiService
- corrected literal errors

tex1988 committed:
Issue#202: ContactService, CooperationService, HouseService, NewsService, UserService
- removed methods for getting one element by id

tex1988 committed:
Issue#202: ContactServiceImpl, CooperationServiceImpl, HouseServiceImpl, NewsServiceImpl, UserServiceImpl
- removed implementation of methods for getting one element by id
- added validation in findAll methods to prevent returning of empty page

tex1988 committed:
Issue#202: Changes in QueryApiService, ContactServiceImpl, CooperationServiceImpl
- changes in getOne in QueryApiService
- checkstyle fixes in ContactServiceImpl and CooperationServiceImpl

tex1988 committed:
Issue#202: Changes in QueryApiService
- changed getOne method and added private getMessage method for more informative NotFoundHomeException message

tex1988 committed:
Issue#202: Changes in ContactServiceImpl, CooperationServiceImpl, HouseServiceImpl, NewsServiceImpl, UserServiceImpl
- returned @Transactional annotation on findAll methods to avoid mapping exceptions

tex1988 committed:
Issue#202: Changes in ContactApiIT, NewsApiIT, UserApiIT
- fixed tests according to new exception messages

tex1988 committed:
Revert "Issue#202: Changes in QueryApiService - changed getOne method and added private getMessage method for more informative NotFoundHomeException message"

This reverts commit 2a9bd1d7

tex1988 committed:
Issue#202: Changes in QueryApiService
- reverted changes getPageFromQuery method
- changes in getOne method according to getPageFromQuery method changes

tex1988 committed:
Issue#202: Changes in ContactServiceImpl, CooperationServiceImpl, HouseServiceImpl, NewsServiceImpl, UserServiceImpl
- reverted all changes in findAll methods

tex1988 committed:
Issue#202: Changes in ContactApiIT, NewsApiIT, UserApiIT
- fixed tests according to new exception messages

tex1988 committed:
Issue#202: Changes in CommonApi, CooperationApiImpl, NewsApiImpl, CooperationServiceImpl
- deleted unused imports and variables

tex1988 committed:
Issue#202: CooperationServiceImpl
- deleted unused imports

kh0ma commented on home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/contacts/ContactApiIT.java:
```diff
@@ -414,7 +414,7 @@ void passNullIdContactWhenDeleteAnyContactTest() throws ApiException {
             .isThrownBy(() -> contactApi
                 .deleteContactOnUserWithHttpInfo(expectedUser.getId(), null))
-            .withMessageContaining("Missing the required parameter 'contactId' when calling deleteContactOnUser");
+            .withMessageContaining("Missing the required parameter 'id' when calling deleteContactOnUser");
```
Isn't it an error that client throws? I mean without any real API call?

tex1988 replied:
Yes, it is but this is not the essence of the task. A separate task will be created to solve this problem.

kh0ma commented on home-application/src/main/java/com/softserveinc/ita/homeproject/application/service/QueryApiService.java:
```diff
@@ -100,6 +103,27 @@ private static void validateQueryParamValue(String queryParamName, List<String>
+        if (page.getTotalElements() == 1) {
+            return page.getContent().get(0);
+        } else if (page.getTotalElements() == 0) {
+            throw new NotFoundHomeException("Entity with id: " + getParameterValue("id", uriInfo)
```
I think that instead of collecting wrong parameters using `getParameterValue("id", uriInfo)` we just can collect it from path parameters. For example:
- query /users/55/contacts/61
- error - "Entity with " + <Use stream or loop to decorate path parameters> + " is not found". ("Entity with 'id = 61 and user_id = 55' is not found")

tex1988 replied:
The proposed changes have been made.

kh0ma commented on home-application/src/main/java/com/softserveinc/ita/homeproject/application/api/CooperationApiImpl.java:
```diff
@@ -67,7 +67,7 @@ public Response createHouse(Long cooperationId, @Valid CreateHouse createHouse)
     @Override
     public Response getCooperation(Long cooperationId) {
-        CooperationDto readCoopDto = cooperationService.getCooperationById(cooperationId);
+        CooperationDto readCoopDto = (CooperationDto) queryApiService.getOne(uriInfo, cooperationService);
```
Can we move this cast into getOne method?

tex1988 replied:
It is impossible for now because of Java behavior which ignores the generic of the method in the case when the generic of the class is not defined. A separate task will be created to refactor QueryApiService and API implementations to avoid of using entity services inside QueryApiService.

tex1988 committed:
Issue#202: ContactService, CooperationService, HouseService, NewsService, UserService
- deleted findAll methods as they inherited from QueryableService

tex1988 committed:
Issue#202: Changes in QueryApiService
- added private getMessage method for more informative exception messages
- changed getOne method to use getMessage method form exception messages

tex1988 committed:
Issue#202: Changes in ContactApiIT, NewsApiIT, UserApiIT
- fixed tests according to the new exception messages

tex1988 committed:
Issue#202: changes in ContactService, CooperationService, HouseService, NewsService, UserService
- deleted unused imports

tex1988 committed:
Issue#202: changes in QueryApiServiceImpl
- EntitySpecificationServiceImpl field changed on EntitySpecificationService

tex1988 committed:
Issue#202: Added QueryApiServiceImplTest

tex1988 committed:
Issue#209: Changes in pom.xml of home-application module
- added jacoco-maven-plugin to get coverage of unit tests

tex1988 committed:
Issue#209: Changes in pom.xml of home-application module
- added maven-surefire-plugin to run unit test on build

tex1988 committed:
Issue#209: Changes in QueryApiServiceImplTest
- removed public modifier from test methods according to Sonar warnings

kh0ma commented on home-application/pom.xml:
```diff
@@ -124,6 +129,14 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
```
Why do we need to deal with this plugin?
